app/scripts/modules/core/src/pipeline/config/stages/StageConfigWrapper.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+
+import { IStageConfigProps } from 'core/pipeline';
+
+export interface IStageConfigWrapperProps extends IStageConfigProps {
+  component: React.ComponentType<any>;
+}
+
+/* This wrapper component exists so that StageConfig components don't have to manually
+ * call this.forceUpdate() after updating a stage field. The only reason it is currently
+ * needed is because the stage config object is managed in angular, above the render root.
+ * When it is eventually migrated to react, it will simply be part of the state of some
+ * parent component and this wrapper can be removed.
+ */
+
+export class StageConfigWrapper extends React.Component<IStageConfigWrapperProps> {
+  public render() {
+    const { component: StageConfig, updateStageField, ...otherProps } = this.props;
```

Comment:
I don't think `component` is used?

Reply:
It destructures `this.props.component` into `StageConfig`, which is used.

Reply:
ahhh, I didn't know you could rename variables in a destructuring call! Thanks!